amp_iframe.html

```diff
@@ -12,39 +12,56 @@
 <body style="padding:0; margin:0; width:100%; height:100%; overflow:hidden">
   <div id='container' style='width:100%; height:100%;'></div>
   <script>
-    function queryStringToJSON(qs) {
-      var pairs = qs.split('&');
-      var result = {};
-      pairs.forEach(function(p) {
-        var pair = p.split('=');
-        var key = pair[0];
-        var value = decodeURIComponent(pair[1] || '');
-        if (value === "true") {
-          value = true;
-        } else if (value == "false") {
-          value = false;
-        } else if (!isNaN(parseFloat(value))) {
-          value = parseFloat(value);
-        } else if (value.match(/^\[.+\]$/)) {
-          value = JSON.parse(value);
-        } else if (value.match(/^\{.+\}$/)) {
-          value = JSON.parse(value);
-        } 
-
-        if (result[key]) {
-          if (Array.isArray(result[key])) {
-            result[key].push(value);
-          } else {
-            result[key] = [result[key], value];
+    function queryStringToJSON(query) {
+      var setValue = function(root, path, value) {
+        if (path.length > 1){
```

Comment:
Path has already been verified by code that calls setValue, this is actually redundant check altogether.